pkg/sql/codegen/feature_column.go

```diff
@@ -66,6 +91,15 @@ type EmbeddingColumn struct {
 	Dimension      int
 	Combiner       string
 	Initializer    string
+	// only used when EMBEDDING(col_name, ...) this will set CategoryColumn = nil
+	// will fill the feature column details using feature_derivation
+	Name string
 }
 
-func (EmbeddingColumn) isFeatureColumn() {}
+// GetFieldMeta returns FieldMeta member
+func (ec *EmbeddingColumn) GetFieldMeta() []*FieldMeta {
+	if ec.CategoryColumn == nil {
```

Comment:
`nil` point is likely to cause panic.

Instead of returning `nil`, why not
- return an empty slice? 
- return `[]*FieldMeta, error`

Reply:
Thanks, I was considering exactly the same thing, will do in next PR.